### ISSUE TITLE
Doesn't log request id when you have multiple threads in the application.

### ISSUE THREAD

**Issue by ajays20078** [dabapps/django-log-request-id#13]
When the Django app creates multiple threads using threading.Thread to handle the incoming requests, it fails to log the request id and gives the following error:

AttributeError: 'thread._local' object has no attribute 'request_id'

**Comment by j4mie** [comment]
This is by design, as the request ID is stored in a thread local. I'm not sure whether it'd be possible to hook in at the point where a new thread is created and copy the property across, or ever whether that'd be a good idea.

Perhaps it shouldn't error - I'd be happy to accept a PR that adds a bit of defensive programming around the attribute access.

**Comment by codingjoe** [comment]
Same issue, here, it actually took our production system down, because the 301 redirects on whether or not you have a `/` (slash) at the end, raised an attribute error.

Anyways. Since it did work before, I'd consider this a bug and fix it, maybe just catch the exception. If you want to raise on missing request ids, this should be changed in a major release.

Here is our stack trace:
```python
AttributeError: request_id
  File "django/core/handlers/base.py", line 235, in get_response
    response = middleware_method(request, response)
  File "newrelic/hooks/framework_django.py", line 335, in wrapper
    return wrapped(*args, **kwargs)
  File "common/middleware.py", line 112, in process_response
    response,
  File "log_request_id/middleware.py", line 42, in process_response
    del local.request_id
```

**Comment by j4mie** [comment]
Ah @codingjoe looks like you've hit a different bug there - that line was only added this week and should be fixed. I'll have a look at that, but in the meantime can you give me a bit more information on your environment? I don't fully understand why this would error in the particular case you describe..

**Comment by codingjoe** [comment]
@j4mie check out my PR and the description, let me know if you are missing any info.

**Comment by j4mie** [comment]
Reopening this because the fix in #13 corrected a problem introduced by #11, which was only merged a few days ago. The original issue must have been elsewhere in the code.

@ajays20078 can you provide any more information on where your error occurred, what web server you're using, etc? Do you have a traceback?

**Comment by luzfcb** [comment]
Is not directly related to this issue, and maybe this is not the best place to ask, but I would like to know what thoughts of this approach https://github.com/Alir3z4/django-cuser/blob/master/cuser/middleware.py#L42-L58 vs the `threading.local()` approach .
